EngineBay.CommunityEdition/Modules/ModuleRegistration.cs

```diff
@@ -79,7 +75,7 @@ public static WebApplication InitializeDatabase(this WebApplication app)
             var serviceProvider = scope.ServiceProvider;
             var dbInitialiser = serviceProvider.GetRequiredService<DbInitialiser>();
 
-            dbInitialiser.Run(RegisteredModules);
+            dbInitialiser.Run(GetRegisteredModules());
```

Comment:
Will repeatedly calling the GetRegisteredModules() method not create a new copy of the list with a new instance of each module each time? Would it be better to assign the module list to a private field after the first call and return that saved list each time?

Reply:
We could introduce a static constructor, but I do not like keeping state in a static class. This class has multiple public entry points that are invoked out of order, meaning each method would need to be made aware of whether the list has been initialized and also never interact with the private list directly.  

This is safer and easier and speed is not really an issue since it only runs at startup.

For context, the order in which the public methods are called is different for generating migrations for SQL Server vs generating migrations for SQL Lite 😱